File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        unsigned short mask = 0;
        for (char i = 0; i < board.size(); i++) {
            for (char j = 0; j < board.size(); j++) {
                if (board[i][j] != '.') {
                    if (mask & (1 << (board[i][j] - '0'))) {
                        return false;
                    }
                    mask |= (1 << (board[i][j] - '0'));
                }
            }
            mask = 0;
        }
        for (char i = 0; i < board.size(); i++) {
            for (char j = 0; j < board.size(); j++) {
                if (board[j][i] != '.') {
                    if (mask & (1 << (board[j][i] - '0'))) {
                        return false;
                    }
                    mask |= (1 << (board[j][i] - '0'));
                }
            }
            mask = 0;
        }
        for (char i = 0; i < board.size(); i += 3) {
            for (char j = 0; j < board.size(); j += 3) {
                for (char k = i; k < i + 3; k++) {
                    for (char l = j; l < j + 3; l++) {
                        if (board[k][l] != '.') {
                            if (mask & (1 << (board[k][l] - '0'))) {
                                return false;
                            }
                            mask |= (1 << (board[k][l] - '0'));
                        }
                    }
                }
                mask = 0;
            }
        }
        return true;
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        mask = 0
        for i in range(len(board)):
            for j in range(len(board)):
                if board[i][j] == '.':
                    continue
                if mask & (1 << int(board[i][j])):
                    return False
                else:
                    mask |= (1 << int(board[i][j]))
            mask = 0
        for i in range(len(board)):
            for j in range(len(board)):
                if board[j][i] == '.':
                    continue
                if mask & (1 << int(board[j][i])):
                    return False
                else:
                    mask |= (1 << int(board[j][i]))
            mask = 0
        for i in range(0, len(board), 3):
            for j in range(0, len(board), 3):
                for k in range(i, i + 3):
                    for l in range(j, j + 3):
                        if board[k][l] == '.':
                            continue
                        if mask & (1 << int(board[k][l])):
                            return False
                        else:
                            mask |= (1 << int(board[k][l]))
                mask = 0
        return True
        